modules/EvseManager/EvseManager.cpp

```diff
@@ -1006,71 +1006,86 @@ void EvseManager::cable_check() {
         session_log.evse(true, "Start cable check...");
         bool ok = false;
 
-        // verify the relais are really switched on and set 500V output
-        if (!contactor_open && powersupply_DC_set(config.dc_isolation_voltage_V, 2)) {
+        // normally contactors should be closed before entering cable check routine.
+        // On some hardware implementation it may take some time until the confirmation arrives though,
+        // so we wait with a timeout here until the contactors are confirmed to be closed.
+        Timeout timeout(CABLECHECK_CONTACTORS_CLOSE_TIMEOUT);
 
-            powersupply_DC_on();
-            imd_start();
+        while (!timeout.reached()) {
+            if (!contactor_open)
+                break;
+            std::this_thread::sleep_for(100ms);
```

Comment:
This timeout highly depend on the used contactor and from our experience sometimes to small. I suggested to use at least 200 ms.

Reply:
That was on purpose, because constants in the class header indicate that they are adjustable if the application requires that, which does not make sense for this one. That is why the timeout is a constant though.